src/PHAL_Utilities.cpp

```diff
@@ -186,10 +186,22 @@ void reduceAll (
   const Teuchos_Comm& comm, const Teuchos::EReductionType reduct_type,
   PHX::MDField<ScalarT>& a)
 {
+  // Old reduceAll implementation is not uvm-friendly and untested
+  //  for rank > 1
+  TEUCHOS_TEST_FOR_EXCEPTION(a.rank() > 1, std::logic_error,
+        "PHAL::reduceAll not implemented for MDFields with rank > 1.\n");
+        
+  MDFieldHostMirror<ScalarT> a_host = Kokkos::create_mirror_view(a.get_view());
+  Kokkos::deep_copy(a_host, a.get_view());
   std::vector<ScalarT> v;
-  copy<ScalarT>(a, v);
+  for (size_t i = 0; i < a.size(); ++i) {
+    v.push_back(a_host(i));
+  }
   myReduceAll<ScalarT>(comm, reduct_type, v);
-  copy<ScalarT>(v, a);
+  for (size_t i = 0; i < a.size(); ++i) {
+    a_host(i) = v[i];
+  }
+  Kokkos::deep_copy(a.get_view(), a_host);
```

Comment:
Is this only used for size == 1 too (global response)? If yes, I think we can remove/simplify a lot of this.

Reply:
Good call, I removed the relevant code that is no longer used or tested.